core/src/main/scala/scalaz/Either.scala

```diff
@@ -397,6 +397,10 @@ object \/ extends DisjunctionInstances {
   def fromEither[A, B](e: Either[A, B]): A \/ B =
     e fold (left, right)
 
+  /** Construct a disjunction value from a standard `scala.Option`. */
+  def fromOption[A, B](o: Option[B], ifNone: => A): A \/ B =
```

Comment:
Personally I think this would be nicer curried as `(ifNone: => A)(o: Option[B])`